pkg/sysregistriesv2/system_registries_v2.go

```diff
@@ -0,0 +1,199 @@
+package sysregistriesv2
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/url"
+	"path/filepath"
+	"strings"
+
+	"github.com/BurntSushi/toml"
+	"github.com/containers/image/types"
+)
+
+// systemRegistriesConfPath is the path to the system-wide registry
+// configuration file and is used to add/subtract potential registries for
+// obtaining images.  You can override this at build time with
+// -ldflags '-X github.com/containers/image/sysregistries.systemRegistriesConfPath=$your_path'
+var systemRegistriesConfPath = builtinRegistriesConfPath
+
+// builtinRegistriesConfPath is the path to the registry configuration file.
+// DO NOT change this, instead see systemRegistriesConfPath above.
+const builtinRegistriesConfPath = "/etc/containers/registries.conf"
+
+// tomlURL is an abstraction required to unmarshal the non-primitive and
+// non-local url.URL type when loading the toml config.
+type tomlURL struct {
+	url url.URL
+}
+
+// Mirror represents a mirror. Mirrors can be used as pull-through caches for
+// registries.
+type Mirror struct {
+	// The mirror's URL
+	URL tomlURL `toml:"url"`
+	// If true, certs verification will be skipped
+	Insecure bool `toml:"insecure"`
+}
+
+// Registry represents a registry.
+type Registry struct {
+	// The registry's URL
+	URL tomlURL `toml:"url"`
+	// If true, pulling from the registry will be blocked
+	Blocked bool `toml:"blocked"`
+	// If true, certs verification will be skipped
+	Insecure bool `toml:"insecure"`
+	// If true, the registry can be used when pulling an unqualified image
+	Search bool `toml:"unqualified-search"`
+	// If it prefixes an image, the registry will be used for pulling
+	// (defaults to the URL without the URI scheme)
+	Prefix string `toml:"prefix"`
```

Comment:
Is the `Prefix` used only for matching, or stripped? E.g. with `URL`=`https://example.com/foo` and `Prefix`=`example.net/bar`, is `example.net/bar/baz` downloaded from `https://example.com/foo/bar/baz` or from `https://example.com/foo/baz` ?

Reply:
The `Prefix` is used only for matching. If we have `URL=https://example.com/foo` and `Prefix=example.net/bar`, then it depends on what the user of the API decides to do. I would suggest downloading `baz:latest` from `https://example.com/foo/baz:latest`. If we want to have the other mapping, we are free to configure `URL=https://example.com/foo/bar` and `Prefix=example.net/bar`.

Reply:
> then it depends on what the user of the API decides to do

I’d much prefer having it explicitly defined. Having two different applications on the same machine interpret the same configuration file differently would not do.

Your example shows stripping the prefix, and you’re right that that gives the user more freedom. So, just explicitly document that, please.